src/header/js/header.js

```diff
@@ -228,7 +250,7 @@ export default (data) => {
 	/* ====================================================================== */
 
 	$header.addEventListener('header:menu:close', ({detail}) => {
-		const currentDetail = detail || searchDetail || accountDetail || appsDetail || notificationsDetail || menusDetail ||  menuDetail;
+		const currentDetail = detail || searchDetail || inlineTitleDetail || accountDetail || appsDetail || notificationsDetail || menusDetail ||  menuDetail;
 
```

Comment:
this is getting a little insane, and how is the order determined here, seems like anything emitting header:menu:close should pass the detail object.

Reply:
Yea, this is one of a few pain points we should look to clean up. There are definitely details that need to take priority to others in this statement. Take a look at line # 172 where we define the different details too.

From what I remember, the `detail` being passed on `header:menu:close` isn't  always the component that needs to be closed. For example, if you click on a menu link, you might get menu link instead of the currently opened component. So we are compensating by saving the current detail on 'header:menu:open'.

I'll create an issue after I've tested out the problem a little more.